hub/rename_data_directories.go

```diff
@@ -92,14 +95,36 @@ func getTargetRenameMap(target InitializeConfig, source *greenplum.Cluster) Rena
 	return m
 }
 
+// IsRenameErrorIdempotent interprets an error returned from os.Rename().  If that error is acceptable, it returns true.
+// The error code options are taken from the Mac OSX manpage and Linux manpage for rename(2).
+//  (These two are consistent.)
+func IsRenameErrorIdempotent(err error) bool {
```

Comment:
So let's say we have the following input case:
```
/data/seg-1_old   <- squatting
/data/seg-1
/data/seg.BLAH.-1
```

The archive to `_old` will fail incorrectly with `EEXIST`, but we'll ignore it. The move to `seg-1` will also fail with `EEXIST`, and we'll ignore that too. Nothing will happen, but we will continue as if everything is okay.

Which error codes are okay depend heavily on which directories are being renamed. (For instance, it seems like we should never, ever, see `EEXIST` during the target->source rename.) Separating the error check from the context using this helper seems like a premature deduplication that's introducing bugs.

Reply:
Good point; I can have separate checks for the source move and the target move.

For your example, "squatting" isn't supposed to happen.  We can check for this explicitly (in a prior substep, unfortunately, since a completed rename has the original source in `old` and the target in source), but our assumption is that some cleanup makes sure the _old doesn't exist.

Perhaps there are better ideas, like moving to `_old_ABC123` as a way of uniquely identifying our ownership; marker files can work too.